renderapi/client.py

```diff
@@ -858,4 +873,254 @@ def transformSectionClient(stack, transformId, transformClass, transformData,
+        This is more general than normalizeForMatching=true, but requires you have transform labels applied.
+        default = None
+    excludeFirstTransformAndAllAfter: str
+        alternative to normalizeForMatching which finds the first transform in the list with a given label
+        and then removes that transform and all transforms that follow it. i.e. if you had a compound list
+        of transformations, and you had labelled the first non-local transform 'montage' then setting
+        excludeFirstTransformAndAllAfter='montage' would remove that montage transform and any other
+        transforms that you had applied after it. default= None.
+    excludeAllTransforms: bool
+        alternative to normalizeForMatching which simply removes all transforms from the list.
+        default=False
+    """
+    request_url = format_preamble(host, port, owner, project, stack)
+    tile_base_url = request_url+"/tile"
+    url_suffix = "render-parameters"
+    if filter:
+        url_suffix += '?filter=true'
+    else:
+        url_suffix += '?filter=false'
+
+    if normalizeForMatching:
+        url_suffix += '&normalizeForMatching=true'
+    else:
+        url_suffix += '&normalizeForMatching=false'
+
+    if renderWithoutMask:
+        url_suffix += '&renderWithoutMask=true'
+    else:
+        url_suffix += '&renderWithoutMask=false'
+
+    if excludeTransformsAfterLast is not None:
+        url_suffix += '&excludeTransformsAfterLast={}'.format(
+            excludeTransformsAfterLast)
+    if excludeFirstTransformAndAllAfter is not None:
+        url_suffix += '&excludeFirstTransformAndAllAfter={}'.format(
+            excludeFirstTransformAndAllAfter)
+    if excludeAllTransforms:
+        url_suffix += '&excludeAllTransforms=true'
+
+    canvas_url_template = "%s/{}/%s" % (tile_base_url,
+                                        url_suffix)
+    return canvas_url_template
+
+
+class ArgumentParameters(object):
+    def __init__(self, *args, **kwargs):
+        pass
+
+    @staticmethod
+    def sanitize_cmd(cmd):
+        def jbool_str(c):
+            return str(c) if type(c) is not bool else "true" if c else "false"
+        if any([i is None for i in cmd]):
+            raise ClientScriptError(
+                'missing argument in command "{}"'.format(map(str, cmd)))
+        return map(jbool_str, cmd)
+
+    @staticmethod
+    def get_cmd_opt(v, flag=None):
+        return [] if v is None else [v] if flag is None else [flag, v]
+
+    @staticmethod
+    def get_flag_cmd(v, flag=None):
+        # for arity 0
+        return [flag] if v else []
+
+    def to_java_args(self):
+        args = []
+        for key, value in self.__dict__.items():
+            if (value is not None) and not (key == 'kwargs'):
+                args += self.get_cmd_opt(value, "--{}".format(key))
+        return self.sanitize_cmd(args)
+
+
+class FeatureRenderParameters(ArgumentParameters):

```

Comment:
This was not actually used, and so you can't render at at lower than scale=1.0

Reply:
err i see it's in there .. but nonetheless this class isn't hooked up to anything.  I think it should go or get incorporated.

Reply:
This conversation should be on the sift client PR, I think.  Things got a little messy with commits on my end but O figured that would get in pretty soon. 
 We can take it out and re-add it when we implement something that uses it....

Reply:
This conversation should be on the sift client PR, I think.  Things got a little messy with commits on my end but O figured that would get in pretty soon. 
 We can take it out and re-add it when we implement something that uses it....

Reply:
ok this PR looks cleaner now after the point match merge